walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,92 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
+
+  /**
+  * Queries through a list of events and creates a list of possible time ranges
+  * where the requested meeting can take place.
+  * @param events the list of events taking place to be considered.
+  * @param request a meeting request with information about the meeting.
+  * @return a list of time ranges where the requested meeting could take place.
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> possibleTimes = new ArrayList<>();
+    List<Event> eventList = new ArrayList<>(events);
+    long duration = request.getDuration();
+
+    // Begins with the assumption that the entire day is available.
+    return findOpennings(possibleTimes, eventList, 0, TimeRange.START_OF_DAY, duration, request.getAttendees());
+  }
+
+  /**
+  * Sorts the list of events taking place by their start times.
+  */
+  private static void sortEventsByStart(List<Event> eventList) {
+    eventList.sort(new Comparator<Event>() {
```

Comment:
I think you can just do something similar to Collections.sort(eventList, TimeRange.ORDER_BY_START);

Reply:
I can't because the comparitor TimeRange.ORDER_BY_START is for TimeRange objects not Event objects. Event doesn't have a comparitor of its own so I think I have to make one.

Reply:
SGTM.